internal/prompt/prompt.go

```diff
@@ -17,23 +18,34 @@ type Prompter interface {
 	Select(title, message string, choices []string, defaultResponse string) (string, *failures.Failure)
 	Confirm(title, message string, defaultChoice bool) (bool, *failures.Failure)
 	InputSecret(title, message string, flags ...ValidatorFlag) (string, *failures.Failure)
+	IsInteractive() bool
 }
 
-// FailPromptUnknownValidator handles unknown validator erros
+// FailPromptUnknownValidator handles unknown validator errors
 var FailPromptUnknownValidator = failures.Type("prompt.unknownvalidator")
 
+// FailPromptNonInteractive handles errors during nonInteractive mode
+var FailPromptNonInteractive = failures.Type("prompt.unknown.nonInteractive")
+
 // ValidatorFunc is a function pass to the Prompter to perform validation
 // on the users input
 type ValidatorFunc = survey.Validator
 
-// Prompt is our main promptig struct
+var _ Prompter = &Prompt{}
+
+// Prompt is our main prompting struct
 type Prompt struct {
-	out output.Outputer
+	out            output.Outputer
+	nonInteractive bool
 }
 
 // New creates a new prompter
-func New() Prompter {
-	return &Prompt{output.Get()}
+func New(nonInteractive bool) Prompter {
+	return &Prompt{output.Get(), nonInteractive}
+}
+
+func (p *Prompt) IsInteractive() bool {
+	return !p.nonInteractive
```

Comment:
Can we just record it as `interactive` here? Negatively phrased booleans always make my brain spin.. :p